src/main/java/com/leverx/learningmanagementsystem/service/impl/CourseServiceImpl.java

```diff
@@ -0,0 +1,130 @@
+        return courseRepository.findAll();
+    }
+
+    @Override
+    @Transactional
+    public Course create(CreateCourseDto createCourseDto) {
+        Course course = courseMapper.toCourse(createCourseDto);
+
+        log.info("Create course with id {}", course.getId());
+        save(course, createCourseDto);
+
+        return course;
+    }
+
+    @Override
+    @Transactional
+    public Course update(UUID id, CreateCourseDto updateCourseDto) {
+
+        if (courseRepository.findById(id).isEmpty()) {
+            throw new EntityNotFoundException("Course with id = " + id + " not found");
+        }
+
+        Course course = courseMapper.toCourse(updateCourseDto);
+        course.setId(id);
+
+        log.info("Update course with id {}", course.getId());
+        save(course, updateCourseDto);
+
+        return course;
+    }
+
+    @Override
+    public List<Course> getAllStartingBetweenDates(LocalDateTime startDate, LocalDateTime endDate) {
+        return courseRepository.findAllStartingTheFollowingDay(startDate, endDate);
+    }
+
+    @Override
+    public void delete(UUID id) {
+        getById(id);
+        log.info("Delete course with id {}", id);
+        courseRepository.deleteById(id);
+    }
+
+    private void save(Course course, CreateCourseDto createCourseDto) {
+        Set<Lesson> lessons;
+        Set<Student> students;
+
+        if (createCourseDto.lessonIds() != null && CollectionUtils.isNotEmpty(createCourseDto.lessonIds())) {
+            log.info("Fetching course's lessons");
+            lessons = new HashSet<>(lessonRepository.findAllById(createCourseDto.lessonIds()));
+            if (lessons.size() != createCourseDto.lessonIds().size()) {
+                throw new IncorrectResultSizeException(
+                        "Some of requested lessons don't exist");
+            }
+        } else {
+            lessons = Collections.emptySet();
+        }
+        if (createCourseDto.studentIds() != null && CollectionUtils.isNotEmpty(createCourseDto.studentIds())) {
+            log.info("Fetching course's students");
+            students = new HashSet<>(studentRepository.findAllById(createCourseDto.studentIds()));
+            if (students.size() != createCourseDto.studentIds().size()) {
+                throw new IncorrectResultSizeException(
+                        "Some of requested students don't exist");
+            }
+        } else {
+            students = Collections.emptySet();
+        }
+
+        UUID settingsId = createCourseDto.courseSettingsId();
+        if (settingsId != null) {
+            log.info("Fetching course settings with id = {}", settingsId);
+            CourseSettings courseSettings = courseSettingsRepository.findById(settingsId)
+                    .orElseThrow(() -> new EntityNotFoundException("Course settings with id = " + settingsId + " not found"));
+            course.setSettings(courseSettings);
+        } else {
+            course.setSettings(null);
+        }
```

Comment:
these if-else statements could be splitted into multiple methods in order to simplify current method + enhance readability of this logic